source/agora/node/Validator.d

```diff
@@ -449,6 +449,21 @@ public class Validator : FullNode, API
         return new AdminInterface(this.config.validator.key_pair, this.clock);
     }
 
+    /// Ditto
+    public override UTXO getValidatorUTXO (in PublicKey pubkey) @trusted nothrow
+    {
+        auto hash = this.enroll_man.validator_set.getEnrolledUTXO(
+                this.ledger.getBlockHeight(), pubkey);
```

Comment:
here we are referring to the block height as 'this.ledger.getBlockHeight()' - but how would this work with a validator that is catching up, and still doesn't have the correct hight?

Reply:
I don't think it could be a big problem because even if there are multiple `RegistryPayload`s in the registry, it will indicate the same address. I hope this explanation makes sense to you. :)

Reply:
> I don't think it could be a big problem because even if there are multiple `RegistryPayload`s for a `PublicKey` in the registry, they will indicate the same address. I hope this explanation makes sense to you. :)

they can actually indicate different addresses, if the validator's IP changed :) However we do authenticate validators during the handshake, so I agree this is not a big problem.